### PR TITLE
Initial support for multiple LLVM versions

### PR DESCRIPTION
**Pull request by sbc100**
See #11362

### CODE REVIEW

File: ChangeLog.md [hunk 1]
```diff
@@ -17,6 +17,8 @@ See docs/process.md for how version tagging works.
 
 Current Trunk
 -------------
+- A simultaneous support for both llvm 12.0 (tot) and llvm 11.0 (soon to be
```
Comment (kripken) [inline ChangeLog.md:20]:
```suggestion
- Simultaneous support for both llvm 12.0 (tot) and llvm 11.0 (soon to be
```

File: ChangeLog.md [hunk 2]
```diff
@@ -17,6 +17,8 @@ See docs/process.md for how version tagging works.
...
+  stable).  See https://github.com/emscripten-core/emscripten/issues/11362
```
Comment (kripken) [inline ChangeLog.md:21]:
```suggestion
  stable). #11362
```

File: tools/shared.py [hunk 3]
```diff
@@ -412,31 +412,29 @@ def fix_js_engine(old, new):
   return new
 
 
-def expected_llvm_version():
+def supported_llvm_versions():
```
Comment (kripken) [inline tools/shared.py:415]: perhaps a comment that they must be in order (as the callers seem to depend on that)?

File: tools/shared.py [hunk 4]
```diff
@@ -412,31 +412,29 @@ def fix_js_engine(old, new):
...
   if get_llvm_target() == WASM_TARGET:
-    return "12.0"
+    return ('11.0', '12.0')
   else:
-    return "6.0"
+    return ("6.0")
```
Comment (kripken) [inline tools/shared.py:419]: maybe `'` here like above, or both with `"`?

**Comment by sbc100** [comment]
How about we issue a different warning... I try an come up with one.

**Comment by sbc100** [comment]
Putting this on hold for a while.

**Pull request closed without merging**